File: backend/swim-admin/src/main/java/com/smartlight/entity/Alarm.java
package com.smartlight.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "alarm")
public class Alarm {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "alarm_code", nullable = false, unique = true, length = 50)
    private String alarmCode;

    @Column(nullable = false)
    private Integer type;

    @Column(nullable = false)
    private Integer level = 2;

    @Column(name = "streetlight_id")
    private Long streetlightId;

    @Column(name = "cabinet_id")
    private Long cabinetId;

    @Column(name = "area_id")
    private Long areaId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    private Integer status = 0;

    @Column(name = "handler_id")
    private Long handlerId;

    @Column(name = "handle_time")
    private LocalDateTime handleTime;

    @Column(name = "handle_remark", length = 500)
    private String handleRemark;

    @Column(name = "alarm_time", nullable = false)
    private LocalDateTime alarmTime;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "streetlight_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler", "area", "cabinet"})
    private Streetlight streetlight;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "area_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Area area;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (alarmTime == null) alarmTime = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() { updatedAt = LocalDateTime.now(); }
}

File: backend-csharp/SmartStreetlight.Api/Models/Entities/Alarm.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace SmartStreetlight.Api.Models.Entities;

[Table("alarm")]
public class Alarm
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Column("alarm_code"), Required, MaxLength(50)]
    public string AlarmCode { get; set; } = "";

    [Required]
    public int Type { get; set; }

    public int Level { get; set; } = 2;

    [Column("streetlight_id")]
    public long? StreetlightId { get; set; }

    [Column("cabinet_id")]
    public long? CabinetId { get; set; }

    [Column("area_id")]
    public long? AreaId { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = "";

    public string? Description { get; set; }

    public int Status { get; set; } = 0;

    [Column("handler_id")]
    public long? HandlerId { get; set; }

    [Column("handle_time")]
    public DateTime? HandleTime { get; set; }

    [Column("handle_remark"), MaxLength(500)]
    public string? HandleRemark { get; set; }

    [Column("alarm_time")]
    public DateTime AlarmTime { get; set; } = DateTime.Now;

    [Column("created_at")]
    [JsonIgnore]
    public DateTime CreatedAt { get; set; } = DateTime.Now;

    [Column("updated_at")]
    [JsonIgnore]
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    [ForeignKey("StreetlightId")]
    public Streetlight? Streetlight { get; set; }

    [ForeignKey("AreaId")]
    public Area? Area { get; set; }
}
